client/src/pages/PublicProfile.js

```diff
@@ -0,0 +1,107 @@
+import { useState, useContext, useEffect } from 'react';
+import styled from 'styled-components';
+import useStyles from '../themes/theme.profile';
+import { useLocation } from 'react-router-dom';
+import axios from 'axios';
+
+import AppBar from '@material-ui/core/AppBar';
+
+import Navbar from '../components/Navbar';
+import Sidebar from '../components/Sidebar/index';
+import CreatePoll from '../components/CreatePoll/index';
+import Avatar from '@material-ui/core/Avatar';
+
+import UserContext from '../context/UserContext';
+ 
+export default function PublicProfile () {   
+    const classes = useStyles();
+    const { userData, setUserData } = useContext(UserContext);
+    const [openCreatePoll, setOpenCreatePoll] = useState(false);
+    const [selectedUser, setSelectedUser] = useState('');
+    const location = useLocation();
+    
+    const userId = location.pathname.split('/')[1];
+
+    useEffect(() => {
+        const getSelectedUser = async () => {
+            let token = localStorage.getItem('auth-token');
+            
+            const userRes = await axios.post(
+                'http://localhost:5000/users/getone',
+                { userId },
+                { headers: { 'x-auth-token': token } },
+            );
+            setSelectedUser(userRes.data);
+        }
+
+        getSelectedUser();
+    }, [userId]);
+
+    if (!userData.user) {
```

Comment:
wrap this logic code block in a useEffect with no dependency. better to do it this way because if React does something weird where it tries to re-render before your `getUser` function finish executing, you could be calling it again and again